frontend/src/pages/connectionTypes/manage/advanced/FileUploadExtensionRow.tsx

```diff
@@ -0,0 +1,89 @@
+import * as React from 'react';
+import {
+  Button,
+  FormHelperText,
+  HelperText,
+  HelperTextItem,
+  InputGroup,
+  InputGroupItem,
+  TextInput,
+} from '@patternfly/react-core';
+import { ExclamationCircleIcon, MinusCircleIcon } from '@patternfly/react-icons';
+import { EXTENSION_REGEX } from '~/pages/connectionTypes/const';
+
+type FileUploadExtensionRowProps = {
+  extension?: string;
+  onChange: (newValue: string) => void;
+  onRemove?: () => void;
+  allowRemove: boolean;
+  textRef?: (ref: HTMLInputElement | null) => void;
```

Comment:
Can this be a standard ref object instead?
```suggestion
  textRef?: React.RefObject<HTMLInputElement>;
```